File: MovieTicketBooking/Movie Booking Java/src/main/java/org/example/service/MovieService.java
package org.example.service;

import lombok.AllArgsConstructor;
import lombok.NonNull;
import org.example.exceptions.NotFoundException;
import org.example.model.Movie;

import java.util.Map;
import java.util.UUID;

@AllArgsConstructor
public class MovieService {

    private final Map<String, Movie> movies;

    public Movie addMovie(@NonNull final String name) {

        String id = UUID.randomUUID().toString();
        Movie movie = new Movie(id, name);
        movies.put(id, movie);
        return movie;
    }

    public Movie getMovie(@NonNull final String id) {

        if (!movies.containsKey(id)) {
            throw new NotFoundException();
        }
        return this.movies.get(id);
    }

}

File: MovieTicketBooking/Movie Booking C#/Services/MovieService.cs
using System;
using MovieTicketBooking.Model;

namespace MovieTicketBooking.Services
{
    public class MovieService
    {
        Dictionary<string, Movie> movies = new();

        public string AddMovie(string name)
        {
            string id = Guid.NewGuid().ToString();
            if(movies.ContainsKey(id))
            {
                throw new Exception("movie with this id already exists");
            }
            movies.Add(id, new Movie(id, name));
            return id;
        }
        public Movie GetMovie(string movieId)
        {
            if(!movies.ContainsKey(movieId))
            {
                throw new Exception("movie with this id not exists");
            }
            return movies.GetValueOrDefault(movieId);
        }
    }
}